src/adapter/src/coord/sequencer/inner/create_continual_task.rs

```diff
@@ -24,6 +43,160 @@ impl Coordinator {
+            .instance_snapshot(cluster_id)
+            .expect("compute instance does not exist");
+
+        let debug_name = self.catalog().resolve_full_name(&name, None).to_string();
+        let optimizer_config = optimize::OptimizerConfig::from(self.catalog().system_config())
+            .override_from(&self.catalog.get_cluster(cluster_id).config.features());
+
+        let view_id = self.allocate_transient_id();
+        let catalog_mut = self.catalog_mut();
+        let sink_id = catalog_mut.allocate_user_id().await?;
+
+        // Put a placeholder in the catalog so the optimizer can find something
+        // for the sink_id.
+        let fake_entry = CatalogEntry {
+            item: CatalogItem::Table(Table {
+                create_sql: Some(create_sql.clone()),
+                desc: desc.clone(),
+                conn_id: None,
+                resolved_ids: resolved_ids.clone(),
+                custom_logical_compaction_window: None,
+                is_retained_metrics_object: false,
+                data_source: TableDataSource::TableWrites {
+                    defaults: Vec::new(),
+                },
+            }),
+            referenced_by: Vec::new(),
+            used_by: Vec::new(),
+            id: sink_id,
+            oid: 0,
+            name: name.clone(),
+            owner_id: *session.current_role_id(),
+            privileges: PrivilegeMap::new(),
+        };
+        catalog_mut.hack_add_ct(sink_id, fake_entry);
+
+        // Build an optimizer for this CONTINUAL TASK.
+        let mut optimizer = optimize::materialized_view::Optimizer::new(
+            Arc::new(catalog_mut.clone()),
+            compute_instance,
+            sink_id,
+            view_id,
+            column_names,
+            non_null_assertions,
+            refresh_schedule.clone(),
+            debug_name,
+            optimizer_config,
+            self.optimizer_metrics(),
+        );
+
+        // Replace our placeholder fake ctes with the real output id, now that
+        // we have it.
+        expr.visit_mut_post(&mut |expr| match expr {
+            HirRelationExpr::Get { id, .. } if *id == Id::Local(LocalId::new(0)) => {
+                *id = Id::Global(sink_id);
+            }
+            _ => {}
+        })?;
+
+        // HIR ⇒ MIR lowering and MIR ⇒ MIR optimization (local and global)
+        let raw_expr = expr.clone();
+        let local_mir_plan = optimizer.catch_unwind_optimize(expr)?;
+        let global_mir_plan = optimizer.catch_unwind_optimize(local_mir_plan.clone())?;
+        // MIR ⇒ LIR lowering and LIR ⇒ LIR optimization (global)
+        let global_lir_plan = optimizer.catch_unwind_optimize(global_mir_plan.clone())?;
+        let (mut df_desc, _df_meta) = global_lir_plan.unapply();
+
+        // Timestamp selection
+        let mut id_bundle = dataflow_import_id_bundle(&df_desc, cluster_id.clone());
+        // TODO(ct): Can't acquire a read hold on ourselves because we don't
+        // exist yet.
```

Comment:
The collection creation happens at the end of this method (before `ship_dataflow`) and taking the read hold happens in the compute controller's `create_dataflow` (though the `Schedule` hack is currently skipping it). I don't think it's necessary to take another read hold in the coordinator, since the current scheme takes read holds in the coordinator only to ensure inputs don't get compacted until the compute controller has installed its own read holds.